ArcadiaTeamsBot/Dialogs/NewRequestDialog.cs

```diff
@@ -1,63 +1,271 @@
+                Type = new CreateRequestTypeDTO
                 {
-                    Prompt = (Activity)MessageFactory.Attachment(attachments),
-                }, cancellationToken);
+                    Id = Convert.ToInt32(formData["TypeId"].ToString()),
+                    RequestTypeFields = createRequestTypeFields
+                }
+            };
+
+            var sendRequestsQuery = new CreateNewServiceDeskRequestCommand(newRequest);
+            await this.mediator.Send(sendRequestsQuery, cancellationToken);
+
+            await promptContext.Context.SendActivityAsync(MessageFactory.Text("New request created successfully"), cancellationToken);
+            return true;
         }
 
-        private static async Task<DialogTurnResult> EndStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        private async Task<DialogTurnResult> InputStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
-            if ((string)stepContext.Result == Back)
+            var requestTypeDTO = (ServiceDeskRequestTypeDTO)stepContext.Options;
+
+            var cardBody = new List<AdaptiveElement>
+            {
+                new AdaptiveTextBlock { Text = requestTypeDTO.Title, Weight = AdaptiveTextWeight.Bolder, Size = AdaptiveTextSize.Large },
+
+                new AdaptiveTextInput { Id = "TypeId", Value = requestTypeDTO.Id.ToString(), IsVisible = false },
+
+                new AdaptiveTextBlock("Title"),
+                new AdaptiveTextInput { Id = "Title", Placeholder = "Title" },
+
+                new AdaptiveTextBlock("Description"),
+                new AdaptiveTextInput { Id = "Description", Placeholder = "Description" },
+
+                new AdaptiveTextBlock("Execution date"),
+                new AdaptiveDateInput { Id = "ExecutionDate" },
+
+                new AdaptiveTextBlock("Priority"),
+                new AdaptiveChoiceSetInput
+                {
+                    Value = "2",
+                    Id = "Priority",
+                    Choices = new List<AdaptiveChoice>
+                    {
+                        new AdaptiveChoice { Title = "Low", Value = "1" },
+                        new AdaptiveChoice { Title = "Default", Value = "2" },
+                        new AdaptiveChoice { Title = "High", Value = "3" }
+                    }
+                }
+            };
+
+            var requestTypeUIFields = this.requestTypeUIFactory.CreateRequestTypeUI(requestTypeDTO).RequestTypeUIFields.ToList();
+
+            foreach (var field in requestTypeUIFields)
             {
-                return await stepContext.BeginDialogAsync(nameof(RequestsTypeDialog), null, cancellationToken);
+                AdaptiveTextBlock textBlock;
+                AdaptiveInput input;
+
+                switch (field.FieldType)
+                {
+                    case RequestTypeUIFieldType.Year:
+                        textBlock = new AdaptiveTextBlock("Enter a year");
```

Comment:
Year of what? A field name should be added somewhere.

Reply:
Added